docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+```
+
+Note that only tags are not searchable in this way. Other fields can
+be searched using a filter:
+
+```
+commercializable_models = [x for x in store.all_models() \
+                           if x.license.contains('allowCommercialUse=Sell')]
+```
+
+#### version() -> str:
+
+Returns the version of the database, currently at `3.2`
+
+#### model_info_by_name(name, base_model, model_type) -> ModelConfigBase:
+
+This method exists to ease the transition from the previous version of
+the model manager, in which `get_model()` took the three arguments
+shown above. This looks for a unique model identified by name, base
+model and model type and returns it.
+
+The method will generate a `DuplicateModelException` if there are more
+than one models that share the same type, base and name. While
+unlikely, it is certainly possible to have a situation in which the
+user had added two models with the same name, base and type, one
+located at path `/foo/my_model` and the other at `/bar/my_model`. It
+is strongly recommended to search for models using `search_by_name()`,
+which can return multiple results, and then to select the desired
+model and pass its ke to `get_model()`.
+
+### Writing model configs to the database
+
+Several methods allow you to create and update stored model config
+records.
+
+#### add_model(key, config) -> ModelConfigBase:
+
+Given a key and a configuration, this will add the model's
+configuration record to the database. `config` can either be a subclass of
+`ModelConfigBase` (i.e. any class listed in `AnyModelConfig`), or a
+`dict` of key/value pairs. In the latter case, the correct
+configuration class will be picked by Pydantic's discriminated union
+mechanism.
+
+If successful, the method will return the appropriate subclass of
+`ModelConfigBase`. It will raise a `DuplicateModelException` if a
+model with the same key is already in the database, or an
+`InvalidModelConfigException` if a dict was passed and Pydantic
+experienced a parse or validation error.
+
+### update_model(key, config) -> AnyModelConfig:
+
+Given a key and a configuration, this will update the model
+configuration record in the database. `config` can be either a
+instance of `ModelConfigBase`, or a sparse `dict` containing the
+fields to be updated. This will return an `AnyModelConfig` on success,
+or raise `InvalidModelConfigException` or `UnknownModelException`
+exceptions on failure.
+
+***TO DO:*** Investigate why `update_model()` returns an
+`AnyModelConfig` while `add_model()` returns a `ModelConfigBase`.
+
+### rename_model(key, new_name) -> ModelConfigBase:
+
+This is a special case of `update_model()` for the use case of
+changing the model's name. It is broken out because there are cases in
+which the InvokeAI application wants to synchronize the model's name
+with its path in the `models` directory after changing the name, type
+or base. However, when using the ModelRecordService directly, the call
+is equivalent to:
```

Comment:
> It is broken out because there are cases in which the InvokeAI application wants to synchronize the model's name with its path in the `models` directory after changing the name, type or base.

I don't understand why this warrants a separate method. Can you elaborate?